### DISCUSSION TITLE
Possibility to revert MicroOS update

### DISCUSSION THREAD

**Discussion by Robert-turbo** [mysticaltech/terraform-hcloud-kube-hetzner#1018] [Q&A]
Hi,

last night after MicroOS update to 20231008 I started to have problems with mounting readwritemany volumes.
Logs showed errors with nfs and version. I had no time so i desiced to use terraform apply few times and draining and deleteing nodes in between.

So, is there possible to restore previous MicroOS version? :) 

@mysticaltech do you have idea how to do it?
I was searching for this in issues and discuccions but did not find any. 

ps. Restarting all nodes did not solve the issue.

**Comment by mysticaltech** [comment]
@Robert-turbo You need to downgrade the nfs-client package with zypper, and freeze the version (via transactional-update shell). Another user just downgraded MicroOS, but that's overkill. Just downgrade the client, see previous versions by searching nfs-client for opensuse tumbleweed. When in transactional-update shell, you can use normal zypper commands. GPT-4 can guide you.

**Comment by Robert-turbo** [comment]
OK

```
transactional-update pkg install -y --oldpackage http://download.opensuse.org/history/20231008/tumbleweed/repo/oss/x86_64/nfs-client-2.6.3-39.4.x86_64.rpm
touch /var/run/reboot-required
```
Worked OK

**Reply by mysticaltech** [reply]
@Robert-turbo Thanks for sharing, so that freezes it, yes?

**Reply by mysticaltech** [reply]
I think you need this to properly freeze the package (so that next upgrade round does not update it):

`zypper addlock nfs-client`

To be executed under transactional-udpate shell, see packer file for an example.

**Comment by mysticaltech** [comment]
Also see https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/1026/files
